plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+        """
+
+        fabric_id_list = []
+
+        for fabric in fabric_locations:
+            site_name = fabric.get("site_name_hierarchy")
+            fabric_type = fabric.get("fabric_type", "fabric_site")
+            # Validate the correct fabric_type given in the playbook
+            self.validate_fabric_type(fabric_type).check_return_status()
+            site_id = self.get_site_id(site_name)
+
+            if fabric_type == "fabric_site":
+                fabric_id = self.get_fabric_site_id(site_name, site_id)
+            else:
+                fabric_id = self.get_fabric_zone_id(site_name, site_id)
+
+            if not fabric_id:
+                self.log("Unable to get the Fabric Id as the give site '{0}' is not fabric site/zone".format(site_name), "WARNING")
+                continue
+
+            fabric_id_list.append(fabric_id)
+
+        return fabric_id_list
+
+    def create_vn_payload(self, vn_detail):
+        """
+        Constructs a payload for a Virtual Network based on the provided details.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            vn_detail (dict): A dictionary containing the details required to create the virtual network.
+        Returns:
+            dict: A dictionary representing the virtual network payload. This includes -
+                - "virtualNetworkName": The name of the virtual network.
+                - "fabricIds" (optional): A list of fabric IDs associated with the fabric locations, if applicable.
+                - "anchoredSiteId" (optional): The ID of the anchored site, if it exists.
+        Description:
+            This function generates a payload necessary for creating a virtual network in the Cisco Catalyst Center.
+            Additionally, if an anchored site name is provided, the function attempts to retrieve the associated
+            site ID. If the site ID is found, it fetches the corresponding fabric ID for the anchored site.
+            The constructed payload, containing the virtual network name and optionally the fabric IDs and anchored
+            site ID, is then returned for further use.
+        """
+
+        fabric_locations = vn_detail.get("fabric_site_locations")
+        vn_name = vn_detail.get("vn_name")
+        vn_payload = {
+            "virtualNetworkName": vn_name,
+        }
+
+        if fabric_locations:
+            fabric_ids = self.get_fabric_ids(fabric_locations)
+
+            if fabric_ids:
+                vn_payload["fabricIds"] = fabric_ids
+
+        site_name = vn_detail.get("anchored_site_name")
+        if site_name:
+            site_id = self.get_site_id(site_name)
+            if not site_id:
+                msg = "Given Anchor site '{0}' not  present in Cisco Catalyst Center.".format(site_name)
+                self.log(msg, "ERROR")
+                return vn_payload
+            try:
```

Comment:
self.log("Anchored site ID found for '{0}': {1}".format(site_name, site_id), "DEBUG")